packages/@aws-cdk/core/lib/bundling.ts

```diff
@@ -121,11 +121,21 @@ export class BundlingDockerImage {
       throw new Error('Failed to extract image ID from Docker build output');
     }
 
-    return new BundlingDockerImage(match[1]);
+    return new BundlingDockerImage(match[1], options.assetHashName);
```

Comment:
How about fingerprinting the `Dockerfile` (`cdk.FileSystem.fingerprint()`) together with the `DockerBuildOptions`?

```suggestion
    const hash = FileSystem.fingerprint(path, { extraHash: JSON.stringify(options) })
    return new BundlingDockerImage(match[1], hash);
```

Reply:
Good idea. I'll run with this.